src/index.tsx

```diff
@@ -49,6 +50,12 @@ const renderApp = () => {
     );
 };
 
+export const reRenderApp = (param: any) => {
```

Comment:
Like our discussion offline, a URL change from `history.push('/' + updatedParam.style);` should be just after `unmount` and before `render` to make sure no style A is rendered when the template is in style B, so let's move this thing in both `template-diag` and `upload-item`